dataplane-webserver/src/backups/mod.rs

```diff
@@ -0,0 +1,429 @@
+    let pods_api: Api<Pod> = Api::namespaced(kube_client.clone(), namespace);
+    let label_selector = format!("cnpg.io/cluster={},cnpg.io/instanceRole=primary", namespace);
+    let params = ListParams::default().labels(&label_selector);
+
+    let pods = pods_api
+        .list(&params)
+        .await
+        .map_err(|e| ErrorInternalServerError(format!("Failed to list pods: {}", e)))?;
+
+    if pods.items.is_empty() {
+        return Ok(BackupResult::Failed(
+            "No primary database pod found".to_string(),
+        ));
+    }
+
+    let pod = &pods.items[0];
+    let pod_name = pod
+        .metadata
+        .name
+        .as_ref()
+        .ok_or_else(|| ErrorInternalServerError("Pod has no name"))?;
+
+    tracing::info!(
+        namespace = %namespace,
+        pod = %pod_name,
+        "Found primary database pod for backup"
+    );
+
+    // Check if temback exists and install if needed
+    let check_cmd = vec!["test", "-f", "/var/lib/postgresql/data/temback"];
+    let attach_params = AttachParams::default()
+        .container("postgres")
+        .stderr(true)
+        .stdout(true)
+        .stdin(false);
+
+    let check_result = pods_api.exec(pod_name, check_cmd, &attach_params).await;
+    if check_result.is_err() {
+        tracing::info!(
+            pod = %pod_name,
+            "temback not found, installing..."
+        );
+
+        // Install temback
+        let install_cmd = vec![
+            "sh", 
+            "-c", 
+            "curl -L https://github.com/tembo-io/temback/releases/download/v0.1.0/temback-v0.1.0-linux-amd64.tar.gz | tar -C /var/lib/postgresql/data --strip-components=1 -zxf - temback-v0.1.0-linux-amd64/temback && chmod +x /var/lib/postgresql/data/temback"
+        ];
+
+        let mut install_output = pods_api
+            .exec(pod_name, install_cmd, &attach_params)
+            .await
+            .map_err(|e| ErrorInternalServerError(format!("Failed to install temback: {}", e)))?;
+
+        // Check for installation errors
+        if let Some(mut stderr) = install_output.stderr() {
+            let mut error_msg = String::new();
+            stderr.read_to_string(&mut error_msg).await.map_err(|e| {
+                ErrorInternalServerError(format!("Failed to read install error output: {}", e))
+            })?;
+
+            if !error_msg.is_empty() {
+                return Ok(BackupResult::Failed(format!(
+                    "Failed to install temback: {}",
+                    error_msg
+                )));
+            }
+        }
+
+        tracing::info!(
+            pod = %pod_name,
+            "Successfully installed temback"
+        );
+    }
+
+    // Execute temback command in the pod
+    let bucket_path_str = format!("{}/{}", bucket_name, bucket_path);
+    let backup_cmd = vec![
+        "/var/lib/postgresql/data/temback",
+        "--name",
+        namespace,
+        "--compress",
+        "--clean",
+        "--bucket",
+        &bucket_path_str,
```

Comment:
Need to add the new switch of `--path` here now that `temback` supports bucket paths.